recipes/containers/lighttpd.yaml

```diff
@@ -12,32 +12,58 @@ depends:
 
 
 buildScript: |
-    # start from scratch
-    rm -rf *
+    install_common() {
+        # start from scratch
+        rm -rf *
 
-    # copy rootfs
-    for dep in "${BOB_DEP_PATHS[@]}" ; do
-       cp -an ${dep}/* .
-    done
+        # copy rootfs
+        for dep in "${BOB_DEP_PATHS[@]}" ; do
+           cp -an ${dep}/* .
+        done
 
-    # some setup is needed
-    mkdir -p run var var/log
-    install -d -m 1777 var/tmp
-    install -d -m 0777  var/log/lighttpd
-    ln -sT ../run var/run
+        # some setup is needed
+        mkdir -p run var var/log
+        install -d -m 1777 var/tmp
+        install -d -m 0777  var/log/lighttpd
+        ln -sT ../run var/run
 
-    # install our version of lighttpd.conf
-    install -m 644 $<<lighttpd/lighttpd.conf>> etc/lighttpd/lighttpd.conf
+        # install our version of lighttpd.conf
+        install -m 644 $<<lighttpd/lighttpd.conf>> etc/lighttpd/lighttpd.conf
 
-    # need a lighttpd user and group
-    cat >etc/passwd <<EOF
+        # need a lighttpd user and group
+        cat >etc/passwd <<EOF
     root:x:0:0:root:/root:/bin/bash
     lighttpd:x:33:33:www-data:/var/www:/usr/sbin/nologin
     EOF
-    cat >etc/group <<EOF
+        cat >etc/group <<EOF
     root:x:0:
     lighttpd:x:33:
     EOF
+    }
 
-packageScript: |
-    cp -a $1/* .
+
+multiPackage:
+    "":
+        buildScript: |
+            install_common
+
+        packageScript: |
+            cp -a $1/* .
+
+    image:
+        buildTools: [mke2fs]
+        buildScript: |
+            install_common
+
+            # See https://systemd.io/PORTABLE_SERVICES.html
+            mkdir -p etc/systemd/system
+            install -m 644 $<<lighttpd/lighttpd.service>> \
+                            etc/systemd/system/lighttpd.service
+            touch etc/machine-id
+            touch etc/os-release
+            touch etc/resolve.conf
+            mkdir -p proc sys dev run tmp var/tmp
+
+        packageScript: |
+            dd if=/dev/zero of=lighttpd.raw bs=1M count=64
+            mkfs.ext4 -d $1 lighttpd.raw
```

Comment:
Why not something like squashfs? This will either waste space or will break if the package size grows.

Reply:
Never used squashfs, I'll have a look.